routes/index.js

```diff
@@ -9,7 +10,57 @@
  */
 module.exports = function routes(app, options, next) {
   app.perform('actions', function (done) {
-    // routes go here
+
+    /**
+     * Returns a route handler for the given type.
+     *
+     * @param {string} type - Event type to get (`Status` or `StatusEvent`)
+     * @returns {function} The async route handler
+     */
+    function statusRouteHandler(type) {
+      const find = (type === 'StatusEvent') ? 'findAll' : 'findOne';
+
+      return async function handler(req, res) {
+        const { pkg, env, version } = req.params;
+
+        if (pkg && env && version) {
+          const status = await app.models[type][find]({ pkg, env, version });
+
+          return res.json(status);
+        }
+
+        if (pkg && env) {
+          const head = await app.models.StatusHead.findOne({ pkg, env });
+          const status = await app.models[type][find]({ pkg, env, version: head.version });
+
+          return res.json(status);
+        }
+
+        res.status(400).json(new Error(`Bad path. ${!pkg && 'Missing package.'} ${!env && 'Missing environment.'}`));
+      };
+    }
+
+    app.routes.get('/status/:pkg/:env/:version?', asynHandler(statusRouteHandler('Status')));
+    app.routes.get('/status-events/:pkg/:env/:version?', asynHandler(statusRouteHandler('StatusEvent')));
```

Comment:
is version really optional on this one?  if you pass it you end up calling `findAll` but if you don't then you call `findOne`.
(haven't actually looked to see what those methods actually do ;) )

Reply:
That's not accurate. If it isn't passed, we call `StatusHead.findOne` to get the head version, and then call `StatusEvent.findAll` or `Status .findOne` with the `pkg`, `env`, and head `version`. If it is passed in, we don't need to get the head version first. I guess this could be refactored to:
```javascript
     return async function handler(req, res) {
        const { pkg, env } = req.params;

        if (pkg && env) {
          const version = req.params.version || await app.models.StatusHead.findOne({ pkg, env }).version;
          const status = await app.models[type][find]({ pkg, env, version });

          return res.json(status);
        }

        res.status(400).json(new Error(`Bad path. ${!pkg && 'Missing package.'} ${!env && 'Missing environment.'}`));
      };
```
Which might make this more clear.